firewood/src/merkle.rs

```diff
@@ -1671,23 +1673,17 @@ impl<S: ShaleStore<Node> + Send + Sync, T> Merkle<S, T> {
             return Ok(Proof(proofs));
         }
 
-        let root_node = self.get_node(root)?;
-
-        let mut nodes = Vec::new();
-
-        let node = self.get_node_by_key_with_callbacks(
-            root_node,
-            key,
-            |node, _| nodes.push(node),
-            |_, _| {},
-        )?;
+        let path_iterator = PathIterator::new(key.as_ref(), self, root)?;
```

Comment:
🤔 I think the idiomatic thing to do here would be to expose a method called something like `fn node_iter_by_key(&self, root, key: Key)`. 

I think this would constrain the way the iterator is created. I think there was also talk of eventually attaching the `root` to `Merkle` so that you don't have to pass it in every time too.

I'm not sure if this really matters if the iterator is never really called outside of this module, but it is is, I think that would be the more convenient way to construct it.

Reply:
And huge nit... but typically, I like to order args such that the most frequently changed come last. So, `merkle` -> `root` -> `key`... 

But as I said, huge nit... and I don't even remember to always follow this.